polygon/heimdall/service_test.go

```diff
@@ -255,9 +255,15 @@ func (suite *ServiceTestSuite) setupSpans() {
 		AnyTimes()
 
 	suite.client.EXPECT().
-		FetchSpan(gomock.Any(), gomock.Any()).
-		DoAndReturn(func(ctx context.Context, id uint64) (*Span, error) {
-			return readEntityFromFile[Span](suite.T(), fmt.Sprintf("%s/span_%d.json", spanTestDataDir, id)), nil
+		FetchSpans(gomock.Any(), gomock.Any(), gomock.Any()).
```

Comment:
btw it might be a good idea to return empty spans for the last page (I think the last 81 spans - 1281 spans % 150) so that they can be fetched via the `FetchSpan` flow - ie our test will then cover both sequential and batched code paths